src/payment/data/sagas.js

```diff
@@ -10,14 +10,19 @@ import {
 // Services
 import * as PaymentApiService from './service';
 
+import { saga as couponSaga, addCouponSuccess, addCouponBegin } from '../coupon';
 
 export function* handleFetchBasket() {
```

Comment:
handleFetchBasket handles the request to get the complete basket state and then the downstream modifications to the redux store.  Therefore, it needs to call into the coupon actions to update the coupon state.